eslint.config.mjs

```diff
@@ -0,0 +1,127 @@
+import eslint from '@eslint/js'
+import vitest from '@vitest/eslint-plugin'
+import importPlugin from 'eslint-plugin-import'
+import jestDom from 'eslint-plugin-jest-dom'
+import react from 'eslint-plugin-react'
+import reactCompiler from 'eslint-plugin-react-compiler'
+import reactHooks from 'eslint-plugin-react-hooks'
+import testingLibrary from 'eslint-plugin-testing-library'
+import globals from 'globals'
+import tseslint from 'typescript-eslint'
+
+export default tseslint.config(
+  {
+    ignores: ['**/dist/'],
+  },
+  eslint.configs.recommended,
+  tseslint.configs.recommended,
+  react.configs.flat.recommended,
+  react.configs.flat['jsx-runtime'],
+  {
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.node,
+        ...globals.es2015,
+      },
+      parser: tseslint.parser,
+      parserOptions: {
+        ecmaVersion: 2018,
+        sourceType: 'module',
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
+    },
+    plugins: {
+      import: importPlugin,
```

Comment:
I didn't know, but does this mean that if we use custom rules, we can't use `importPlugin.flatConfigs.recommended`, can we?

Reply:
https://github.com/import-js/eslint-plugin-import/blob/main/config/flat/recommended.js

@dai-shi I considered using `importPlugin.flatConfigs.recommended`, but since there are settings we would need to set `off` and others that would be overridden, I decided to configure it as it is now. I felt it would be better to avoid having `off` settings. The behavior remains the same as before.”

Reply:
that also makes sense. i'll see how it goes. thanks for your suggestion.